File: C_plus_plus/Length_of_Last_Word.cpp
class Solution {
public:
    int lengthOfLastWord(const char *s) {
        int len = 0,last = 0;
        while(*s!='\0'){
            len = 0;
            while(*s!='\0' && *s!=' '){
                ++len;
                ++s;
            }
            if(len) last = len;
            if(*s++ == '\0') break;
        }
        return last;
    }
};

File: java/Length_of_Last_Word.java
// Be careful of the cases : "a    b   ", "a   "
// Time is O(n), space is O(1)

public class Solution {
    public int lengthOfLastWord(String s) {
        int len = 0, last = 0;
        for (int i = 0; i < s.length(); ++i) {
            if (s.charAt(i) == ' ') {
                last = len == 0 ? last : len;
                len = 0;
            }
            else {
                len++;
            }
        }
        return len == 0 ? last : len;
    }
}
